packages/core/src/fields/types/select/index.ts

```diff
@@ -51,67 +51,69 @@ export type SelectFieldConfig<ListTypeInfo extends BaseListTypeInfo> =
     validation,
-    ...config
-  }: SelectFieldConfig<ListTypeInfo>): FieldTypeFunc<ListTypeInfo> =>
-  meta => {
-    const fieldLabel = config.label ?? humanize(meta.fieldKey)
-    const resolvedIsNullable = getResolvedIsNullable(validation, config.db)
-    assertReadIsNonNullAllowed(meta, config, resolvedIsNullable)
+  } = config
 
-    const commonConfig = (
-      options: readonly { value: string | number, label: string }[]
-    ): CommonFieldConfig<ListTypeInfo> & {
-      __ksTelemetryFieldTypeName: string
-      views: string
-      getAdminMeta: () => AdminSelectFieldMeta
-    } => {
-      const values = new Set(options.map(x => x.value))
-      if (values.size !== options.length) {
-        throw new Error(
-          `The select field at ${meta.listKey}.${meta.fieldKey} has duplicate options, this is not allowed`
-        )
+  return (meta) => {
+    const options = config.options.map(option => {
+      if (typeof option === 'string') {
+        return {
+          label: humanize(option),
+          value: option,
+        }
       }
-      return {
-        ...config,
-        ui,
-        hooks: {
-          ...config.hooks,
-          async validateInput (args) {
-            const value = args.resolvedData[meta.fieldKey]
-            if (value != null && !values.has(value)) {
-              args.addValidationError(`${value} is not a possible value for ${fieldLabel}`)
-            }
-            if (
-              (validation?.isRequired || resolvedIsNullable === false) &&
-              (value === null || (value === undefined && args.operation === 'create'))
-            ) {
-              args.addValidationError(`${fieldLabel} is required`)
-            }
-            await config.hooks?.validateInput?.(args)
-          },
-        },
-        __ksTelemetryFieldTypeName: '@keystone-6/select',
-        views: '@keystone-6/core/fields/types/select/views',
-        getAdminMeta: () => ({
-          options,
-          type: config.type ?? 'string',
-          displayMode: displayMode,
-          defaultValue: defaultValue ?? null,
-          isRequired: validation?.isRequired ?? false,
-        }),
+      return option
+    })
+
+    const accepted = new Set(options.map(x => x.value))
+    if (accepted.size !== options.length) {
+      throw new Error(`${meta.listKey}.${meta.fieldKey}: duplicate options, this is not allowed`)
+    }
+
+    const {
+      mode,
+      validate,
+    } = makeValidateHook(meta, config, ({ resolvedData, operation, addValidationError }) => {
+      if (operation === 'delete') return
+
+      const value = resolvedData[meta.fieldKey]
+      if (value != null && !accepted.has(value)) {
+        addValidationError(`value is not an accepted option`)
       }
+    })
+
+    const commonConfig = {
+      ...config,
+      mode,
+      ui,
+      hooks: {
+        ...config.hooks,
+        validate: {
+          ...config.hooks?.validate,
+          create: merge(config.hooks?.validate?.create, validate),
```

Comment:
@dcousens need help with the type here

also can you rebase `v-next` branch to `main`, it does not have anything merge I believe

Reply:
Rebased `v-next` on `main`